### PR TITLE
closes #4545 add AIX to default_settings_yml

### LINKED ISSUE #4545: Add AIX to default_settings_yml
Add AIX to os in default settings.yml.

### PR DESCRIPTION
**Pull request by scddev**
Changelog: Feature: Add AIX to supported OS.
Docs: missing

- [x] fixes #4545
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.

### CODE REVIEW

File: conans/client/tools/oss.py [hunk 1]
```diff
@@ -71,6 +71,14 @@ def detected_architecture():
     elif "arm" in machine:
         return "armv6"
 
+    # processor reports powerpc on AIX
+    processor = platform.processor()
+    if "powerpc" in processor and OSInfo().is_aix:
+        # query bitness by using getconf
+        kernel_bitness = OSInfo().getconf("KERNEL_BITMODE")
```
Comment (scddev) [inline conans/client/tools/oss.py]: currently detect_architecture relies on the bitness of python and not on bitness of the system its running on, so probably I will change it to that behavior too

**Comment by scddev** [comment]
Yes, the problem is the creation of static libs. ar/ranlib require some flags to work. I am still not sure if I would like to go that route. The other option would be to handle that in the conanfile.py of the project by configuring AutoTools or providing a toolchain for cmake. We are currently evaluating that approach.

File: conans/client/conf/__init__.py [hunk 2]
```diff
@@ -12,13 +12,13 @@
...
-arch_build: [x86, x86_64, ppc32, ppc64le, ppc64, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x]
+os_build: [Windows, WindowsStore, Linux, Macos, FreeBSD, SunOS, AIX]
+arch_build: [x86, x86_64, ppc32, ppc64le, ppc64, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x, powerpc, rs6000]
 
 # Only for building cross compilation tools, 'os_target/arch_target' is the system for
 # which the tools generate code
-os_target: [Windows, Linux, Macos, Android, iOS, watchOS, tvOS, FreeBSD, SunOS, Arduino]
-arch_target: [x86, x86_64, ppc32, ppc64le, ppc64, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x]
+os_target: [Windows, Linux, Macos, Android, iOS, watchOS, tvOS, FreeBSD, SunOS, AIX, Arduino]
+arch_target: [x86, x86_64, ppc32, ppc64le, ppc64, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, sparc, sparcv9, mips, mips64, avr, s390, s390x, powerpc, rs6000]
```
Comment (lasote) [inline conans/client/conf/__init__.py]: I think the ppc32, ppc64.. etc already meant PowerPC

Comment (scddev) [reply conans/client/conf/__init__.py]: Basically yes but python and uname show different values for ppc on Linux and AIX:

Linux:
```
~$ uname -a
Linux xxx 2.6.32-754.el6.ppc64 #1 SMP Thu May 24 18:29:49 EDT 2018 ppc64 ppc64 ppc64 GNU/Linux
~$ uname -s
Linux
~$ uname -p
ppc64
~$ uname -m
ppc64
```

AIX:
```
~$ uname -a
AIX xxx 1 7 00FB91F44C00
~$ uname -s
AIX
~$ uname -p
powerpc
~$ uname -m
00FB91F44C00
```

Thats why I added rs6000/powerpc. Previously I already was using ppc32/ppc64. Then I had a look at the AutoTools and config.guess which generates the gnu triplets like:

Linux:
```
~$ ./config.guess 
powerpc64-unknown-linux-gnu
```

AIX:
```
~$ ./config.guess
powerpc-ibm-aix7.1.3.0
```

Comment (lasote) [reply conans/client/conf/__init__.py]: We need to keep the same meaning for the settings, and maybe it was not the best name, but yes, use the ppcxx notation.

File: conans/client/conf/detect.py [hunk 3]
```diff
@@ -38,11 +38,19 @@ def _gcc_compiler(output, compiler_exe="gcc"):
...
         # Since GCC 7.1, -dumpversion return the major version number
         # only ("7"). We must use -dumpfullversion to get the full version
         # number ("7.1.1").
         if installed_version:
+            if not "." in installed_version and int(installed_version) >= 7:
+                ret, out = _execute('%s -dumpfullversion' % compiler_exe)
+                if ret != 0:
+                    return None
+
+            installed_version = re.search("([0-9](\.[0-9])?)", out).group()
```
Comment (lasote) [inline conans/client/conf/detect.py]: What is that about? Specific for AIX? different output there or..? If it is not related please remove it and open a new PR with the "refactors".

File: conans/client/conf/detect.py [hunk 4]
```diff
@@ -169,18 +177,11 @@ def _detect_compiler_version(result, output, profile_path):
 
 
 def _detect_os_arch(result, output):
-    architectures = {'i386': 'x86',
```
Comment (lasote) [inline conans/client/conf/detect.py]: Same comment as above, if it is not related with the AIX we need you to open a different PR. I can see here some changing behavior like the  `sun4v` is not well managed anymore.

File: conans/client/tools/oss.py [hunk 5]
```diff
@@ -304,6 +325,18 @@ def uname(options=None):
         except Exception:
             return None
 
+    @staticmethod
+    def getconf(options=None):
```
Comment (lasote) [inline conans/client/tools/oss.py]: If only for AIX I would call it `aix_get_conf` or similar.

File: conans/client/conf/detect.py (hunk 3 above)
Comment (scddev) [reply conans/client/conf/detect.py]: Ahh ... thats related to an other problem. I will have a look if there is already a bug filed and move it to a separate PR.

```
~$ gcc -dumpversion
7
~$ gcc -dumpfullversion
7.3.0
```

gcc version detection is not working anymore correctly with newer gcc.

Not sure how fast I can update the PR as I am on a tight schedule at the moment. Hopefully I can resolve it soon.

Comment (lasote) [reply conans/client/conf/detect.py]: I have gcc 7 and the detection is working just fine. For gcc >= 5 we are taking only the major version, as they changed the versioning scheme also.

File: conans/client/conf/detect.py (hunk 4 above)
Comment (scddev) [reply conans/client/conf/detect.py]: this is a duplication of the code in detected_architecture which leads to different behavior between those methods or duplicate code. sun4v is the only thing missing in the detected_architecture handling which i will try to add there

Comment (lasote) [reply conans/client/conf/detect.py]: Yes, we know and understand but please open a new PR with this refactor, we agree that it is pretty duplicated but the refactor is changing the behavior and it is not needed for supporting the AIX setting. As I said, for example `sun4v` key is not considered in the `detected_architecture`. We always try to keep the PRs as lean as possible, we have a compatibility commitment and we have to try hard not to change any possible breaking behavior.

**Comment by memsharded** [comment]
Very nice and complete contribution, thanks very much @scddev 

Merged, it will be released in Conan 1.16.

**Pull request merged**